File: 242. Valid Anagram/242.js
// 242. Valid Anagram solution in JavaScript
// https://leetcode.com/problems/valid-anagram/description/

/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    if (s.length != t.length) return false;

    const count = {};

    for (let c of s) {
        count[c] = (count[c] || 0) + 1;
    }

    for (let c of t) {
        if (!count[c]) return false;

        count[c]--;
    }

    return true;
};

File: 242. Valid Anagram/242.go
// 242. Valid Anagram solution in GoLang
// https://leetcode.com/problems/valid-anagram/description/

func isAnagram(s string, t string) bool {
    if len(s) != len(t) {
        return false
    }

    string_map := make(map[rune]int)

    for _, char_s := range s {
        string_map[char_s]++
    }

    for _, char_t := range t {
        string_map[char_t]--
    }

    for _, count := range string_map {
        if count != 0 {
            return false
        }
    }

    return true
}